### ISSUE TITLE
Checkov should not warn about downloading external modules if `--download-external-modules` is explicitly set to `false`.

### ISSUE THREAD

**Issue by bgshacklett** [bridgecrewio/checkov#5782]
**Describe the issue**
When `--download-external-modules` is explicitly set to `false`, Checkov still produces warnings indicating that downloading external modules has failed.

If I am explicitly setting this value to `false`, it is not useful to produce a warning, and may lead to confusion for those running the output.

**Examples**
```
# command
checkov \
  --dir . \
  --download-external-modules false \
  --baseline .checkov.baseline \
  --output-baseline-as-skipped \
  --output cli \
  --output junitxml \
  --output-file-path console,'test-results/checkov.xml'
```

Warning (sanitized):
```
2023-11-21 14:01:42,192 [MainThread  ] [WARNI]  Failed to download module <module_path> (for external modules, the --download-external-modules flag is required)
```

**Version (please complete the following information):**
 - Checkov Version 3.1.8

**Additional context**
Downloading external modules is disabled in our pipeline to avoid performing checks against code that we do not have any control over as part of the CI process.

**Comment by gruebel** [comment]
hey @bgshacklett thanks for reaching out.

I don't think we can actually differentiate between setting the flag with `false` or not, because it is the default. I'll keep it open, if someone wants to investigate it further.

**Comment by kstevensonnv** [comment]
Hi,

Same scenario as raised, I am explicitly turning off checking modules as I have no control over the code.
No need to warn with 20+ lines that an external module is not being downloaded.

**Comment by venkat-raman-wday** [comment]
Same here. It's counterintuitive to set the the flag to false only for a checkov run to fail. I'm seeing this on version 3.2.22.

**Comment by bgshacklett** [comment]
Still an open issue

**Comment by ChananM** [comment]
Hi @bgshacklett, thank you for reaching out
Unfortunately we won't be able to address this issue in the foreseeable future.
However, we are [accepting contributions](https://github.com/bridgecrewio/checkov?tab=readme-ov-file#contributing), and would be happy to review a solution suggested by the community.
